hypershift-operator/main.go

```diff
@@ -434,3 +439,148 @@ func run(ctx context.Context, opts *StartOptions, log logr.Logger) error {
 	log.Info("starting manager")
 	return mgr.Start(ctx)
 }
+
+func NewSetupCommand() *cobra.Command {
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true), zap.JSONEncoder(func(o *zapcore.EncoderConfig) {
+		o.EncodeTime = zapcore.RFC3339TimeEncoder
+	})))
+
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Initializes the environment for the Hypershift Operator",
+	}
+
+	cmd.Run = func(cmd *cobra.Command, args []string) {
+		ctx, cancel := context.WithCancel(ctrl.SetupSignalHandler())
+		defer cancel()
+
+		if err := runInit(ctx, ctrl.Log.WithName("init")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	}
+
+	return cmd
+}
+
+func runInit(ctx context.Context, log logr.Logger) error {
+	log.Info("Initializing environment for Hypershift Operator")
+	client, err := util.GetClient()
+	if err != nil {
+		return err
+	}
+	config := ctrl.GetConfigOrDie()
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(config)
+	if err != nil {
+		return fmt.Errorf("unable to get discovery client: %w", err)
+	}
+	mgmtClusterCaps, err := capabilities.DetectManagementClusterCapabilities(discoveryClient)
+	if err != nil {
+		return fmt.Errorf("unable to detect management cluster capabilities: %w", err)
+	}
+
+	if mgmtClusterCaps.Has(capabilities.CapabilityImage) && mgmtClusterCaps.Has(capabilities.CapabilityProxy) {
+		imageRegistryCABundle, err := getImageRegistryCABundle(client)
+		if err != nil {
+			return err
+		}
+		if imageRegistryCABundle != "" {
+			// Create a new ConfigMap containing all the image registry CA certificates as a bundle
+			// Proxy expects the ConfigMap to:
+			// 1. contain only "ca-bundle.crt" as the key
+			// 2. exist in the openshift-config namespace
+			imageRegistryCABundleCM := &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "v1",
+					Kind:       "ConfigMap",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "user-image-registry-ca-bundle",
+					Namespace: "openshift-config",
+				},
+				Data: map[string]string{
+					"ca-bundle.crt": imageRegistryCABundle,
+				},
+			}
+			if err := apply(ctx, client, imageRegistryCABundleCM); err != nil {
+				return err
+			}
+
+			// Adding the ConfigMap containing the CA bundle to the Proxy object
+			// will allow the openshift-network-operator to automatically
+			// inject this bundle into the openshift-config-managed-trusted-ca-bundle ConfigMap (generated
```

Comment:
>// inject this bundle into the openshift-config-managed-trusted-ca-bundle ConfigMap (generated
			// during hypershift install command)

Can you link to where install is doing this?

Reply:
https://github.com/openshift/hypershift/blob/80dcc89840817fe96b9b9c738b5707eef8026248/cmd/install/install.go#L483-L490

Reply:
Sorry I may have misunderstood - did you mean to put the link to it as a code comment?